src/raftstore/store/peer.rs

```diff
@@ -1875,6 +1800,124 @@ impl Peer {
     }
 }
 
+/// `RequestPolicy` decides how we handle a request.
+#[derive(Clone, PartialEq, Debug)]
+enum RequestPolicy {
+    // Handle the read request directly without dispatch.
+    ReadLocal,
+    // Handle the read request via raft's SafeReadIndex mechanism.
+    ReadIndex,
+    ProposeNormal,
+    ProposeTransferLeader,
+    ProposeConfChange,
+}
+
+/// A `RequestCondition` helps `RequestInspector` make a `RequestPolicy`.
+trait RequestCondition {
```

Comment:
Sorry, I can't connect the name "condition" with these methods, how about `RequestHandler`?

Reply:
What will implement this trait?

Reply:
`Peer`, but it does not mean this trait is supposed to to execute or handle requests. You know traits are independent to their implementer. I can define a new struct who implements the trait.

```rust
struct Foo {
  appiled_index_term,
  lease,
}
```